controllers/persistentvolumeclaim_controller.go

```diff
@@ -0,0 +1,260 @@
+		logger.Error(err, "Failed to create reclaimSpaceCronJob")
+
+		return ctrl.Result{}, err
+	}
+
+	// add cronjob name in annotations.
+	patch := []byte(fmt.Sprintf(`{"metadata":{"annotations":{%q:%q}}}`, rsCronJobNameAnnotation, rsCronJobName))
+	err = r.Client.Patch(ctx, pvc, client.RawPatch(types.StrategicMergePatchType, patch))
+	if err != nil {
+		logger.Error(err, "Failed to update annotation")
+
+		return ctrl.Result{}, err
+	}
+
+	logger.Info("Successfully created reclaimSpaceCronJob")
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *PersistentVolumeClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	err := mgr.GetFieldIndexer().IndexField(
+		context.Background(),
+		&csiaddonsv1alpha1.ReclaimSpaceCronJob{},
+		jobOwnerKey,
+		extractOwnerNameFromPVCObj)
+	if err != nil {
+		return err
+	}
+
+	pred := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			if e.Object == nil {
+				return false
+			}
+			// reconcile only if schedule annotation is found.
+			_, ok := e.Object.GetAnnotations()[rsCronJobScheduleTimeAnnotation]
+			return ok
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if e.ObjectNew == nil || e.ObjectOld == nil {
+				return false
+			}
+			// reconcile only if schedule annotation between old and new objects have changed.
+			oldSchdeule, oldOk := e.ObjectOld.GetAnnotations()[rsCronJobScheduleTimeAnnotation]
+			newSchdeule, newOk := e.ObjectNew.GetAnnotations()[rsCronJobScheduleTimeAnnotation]
+			return oldOk != newOk || oldSchdeule != newSchdeule
+		},
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.PersistentVolumeClaim{}).
+		WithEventFilter(pred).
+		Complete(r)
+}
+
+// deleteChildCronJob lists and deletes all child cron job.
+func (r *PersistentVolumeClaimReconciler) deleteChildCronJob(
+	ctx context.Context,
+	logger *logr.Logger,
+	req *reconcile.Request) error {
+	var childJobs csiaddonsv1alpha1.ReclaimSpaceCronJobList
+	err := r.List(ctx,
+		&childJobs,
+		client.InNamespace(req.Namespace),
+		client.MatchingFields{jobOwnerKey: req.Name})
+	if err != nil {
+		logger.Error(err, "Failed to list child ReclaimSpaceCronJobs")
+
+		return err
+	}
+
+	for _, job := range childJobs.Items {
+		err = r.Delete(ctx, &job)
+		if client.IgnoreNotFound(err) != nil {
+			logger.Error(err, "Failed to delete child ReclaimSpaceCronJob",
+				"reclaimSpaceCronJobName", job.Name)
+
+			return err
+		}
+	}
+
+	return nil
+}
+
+// getScheduleFromAnnotation parses schedule and returns it.
+// A error is logged and default schedule is returned if it
+// is not in cron format.
+func getScheduleFromAnnotation(
+	logger *logr.Logger,
+	annotations map[string]string) (string, bool) {
+	schedule, ok := annotations[rsCronJobScheduleTimeAnnotation]
+	if !ok {
+		return "", false
+	}
+	_, err := cron.ParseStandard(schedule)
+	if err != nil {
+		logger.Info(fmt.Sprintf("Parsing given schedule %q failed, using default schedule %q",
```

Comment:
Sounds like logger.Warning?

Reply:
> Sounds like logger.Warning?

Should be fine for now, there is no default logger.Warning, 
We may add it the future.